inference/enterprise/workflows/enterprise_steps/sinks/opc_writer/v1.py

```diff
@@ -0,0 +1,367 @@
+
+* numbers (integers, floats)
+* booleans
+* strings
+
+Type of sent data must match type of OPC object.
+
+### Cooldown
+
+The block accepts `cooldown_seconds` (which **defaults to `5` seconds**) to prevent unintended bursts of 
+traffic sent to OPC server. Please adjust it according to your needs, setting `0` indicate no cooldown. 
+
+During cooldown period, consecutive runs of the step will cause `throttling_status` output to be set `True`
+and no data will be sent.
+
+### Async execution
+
+Configure the `fire_and_forget` property. Set it to True if you want the data to be sent in the background, 
+allowing the Workflow to proceed without waiting on data to be sent. In this case you will not be able to rely on 
+`error_status` output which will always be set to `False`, so we **recommend setting the `fire_and_forget=False` for
+debugging purposes**.
+
+### Disabling notifications based on runtime parameter
+
+Sometimes it would be convenient to manually disable the **OPC Writer** block. This can be achieved by
+setting `disable_sink` flag to hold reference to Workflow input. With such setup, caller cat disable the sink
+by sending agreed input parameter.
+
+!!! warning "Cooldown limitations"
+    Current implementation of cooldown is limited to video processing - using this block in context of a 
+    Workflow that is run behind HTTP service (Roboflow Hosted API, Dedicated Deployment or self-hosted 
+    `inference` server) will have no effect with regards to cooldown timer.
+"""
+
+QUERY_PARAMS_KIND = [
+    STRING_KIND,
+    INTEGER_KIND,
+    FLOAT_KIND,
+    ROBOFLOW_MODEL_ID_KIND,
+    ROBOFLOW_PROJECT_KIND,
+    ROBOFLOW_API_KEY_KIND,
+    FLOAT_ZERO_TO_ONE_KIND,
+    LIST_OF_VALUES_KIND,
+    BOOLEAN_KIND,
+    TOP_CLASS_KIND,
+]
+HEADER_KIND = [
+    STRING_KIND,
+    INTEGER_KIND,
+    FLOAT_KIND,
+    ROBOFLOW_MODEL_ID_KIND,
+    ROBOFLOW_PROJECT_KIND,
+    ROBOFLOW_API_KEY_KIND,
+    FLOAT_ZERO_TO_ONE_KIND,
+    BOOLEAN_KIND,
+    TOP_CLASS_KIND,
+]
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "OPC Writer Sink",
+            "version": "v1",
+            "short_description": "Pushes data to OPC server, this block is making use of [asyncua](https://github.com/FreeOpcUa/opcua-asyncio)",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "sink",
+        }
+    )
+    type: Literal[BLOCK_TYPE]
+    url: Union[Selector(kind=[STRING_KIND]), str] = Field(
+        description="URL of OPC server where data should be pushed to",
+        examples=["$inputs.opc_url", "opc.tcp://localhost:4840/freeopcua/server/"],
+    )
+    namespace: Union[Selector(kind=[STRING_KIND]), str] = Field(
+        description="OPC namespace",
+        examples=["$inputs.opc_namespace", "http://examples.freeopcua.github.io"],
+    )
+    user_name: Union[Selector(kind=[STRING_KIND]), Optional[str]] = Field(
```

Comment:
not sure how UI will render that, we were always doing `Optional[Union[...]]` - typing wise, it's ok just needs to be checked

Reply:
the same for pw